### ISSUE TITLE
FR: lint error for 'Can't take value of a macro'

### ISSUE THREAD

**Issue by nottmey** [clj-kondo/clj-kondo#1725]
To upvote this issue, give it a thumbs up. See [this list](https://github.com/clj-kondo/clj-kondo/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc) for the most upvoted issues.

**Is your feature request related to a problem? Please describe.**
When coding I don't always realize whether I'm using a function or a macro as a value.

Example: 
```clojure
; incorrect, causes error "Can't take value of a macro: #'clojure.core/or"
(update m :k or "value I want to have set, if :k has no value")

; correct
(update m :k #(or % "value I want to have set, if :k has no value"))
```

**Describe the solution you'd like**
I would like to have a linter error on the macro symbol (`or` in the example), so that I realize, that I'm using a macro as a value, when writing the code.

**Describe alternatives you've considered**
(Alternative is fast feedback cycles - REPL yay - to stumble upon this error as early as possible 😄)

**Additional context**

**Comment by borkdude** [comment]
I'm working on this in a branch called `cant-take-value-of-macro` but I'm getting a lot of false positives currently which I'll have to address first. A general problem is when a user-defined macro refers to a macro by name as a value, which can cause a false positive.

**Comment by borkdude** [comment]
TODO: references to macros as vars

- [ ] `#'pprint/with-pretty-writer`
